airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -277,12 +277,12 @@ async def run(self) -> AsyncIterator[TriggerEvent]:
                     self.polling_interval_seconds,
                 )
                 await asyncio.sleep(self.polling_interval_seconds)
-            except NotFound:
-                yield TriggerEvent({"status": "success", "message": ""})
-                return
-            except Exception as e:
-                yield TriggerEvent({"status": "error", "message": str(e)})
-                return
+        except NotFound:
+            yield TriggerEvent({"status": "success", "message": ""})
+            return
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e)})
+            return
         yield TriggerEvent({"status": "error", "message": "Timeout"})
```

Comment:
```suggestion
        except NotFound:
            yield TriggerEvent({"status": "success", "message": ""})
        except Exception as e:
            yield TriggerEvent({"status": "error", "message": str(e)})
        else:
            yield TriggerEvent({"status": "error", "message": "Timeout"})
```

Reply:
Good idea, I kept the return statements to avoid yielding two events, but the suggested else could help.